### PR TITLE
Heat map chart unit tests

### PR DESCRIPTION
**Pull request by v-baambati**
1.Added unit test cases
Code coverage before PR
HeatMapChart_before

Code coverage After PR
Heatmap-Final

### CODE REVIEW

**Comment by AtishayMsft** [comment]
@krkshitij Could you review this PR for heatmap.

File: packages/react-charting/docs/TestPlans/HeatMapChart/UnitTests.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
+**[Unit Test Plan -- Heat Map Chart]{.underline}**
```
[the remark this answers is not in the document]
Comment (v-baambati) [reply packages/react-charting/docs/TestPlans/HeatMapChart/UnitTests.md]: Updated.

File: packages/react-charting/UnitTests/HeatMapChartUT.test.tsx [hunk 2]
```diff
@@ -0,0 +1,676 @@
...
+
+  test('Should return proper xIndex for date xPoint', () => {
+    const instance = new HeatMapChartBase({
+      data: HeatMapDataNumaricPoints,
+      domainValuesForColorScale: domainValuesForColorScale,
+      rangeValuesForColorScale: rangeValuesForColorScale,
+    });
+    expect(instance).toBeDefined();
+    const xIndex = instance._getXIndex(xPoint2[0]);
+    expect(xIndex).toEqual('Fri Apr 03 2020 05:30:00 GMT+0530 (India Standard Time)');
```
Comment (AtishayMsft) [inline packages/react-charting/UnitTests/HeatMapChartUT.test.tsx:223]: This test will break if run for other time zones. Put it under the skip category.

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/9ae4e5d426eb1a88f56388f620a4265b1302c664'>9ae4e5d426eb1a88f56388f620a4265b1302c664</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-charting/UnitTests/HeatMapChartUT.test.tsx [hunk 3]
```diff
@@ -0,0 +1,676 @@
...
+
+runTest('_getStringFormattedDate', () => {
+  test('Should return proper string formatted date for date point', () => {
+    const instance = new HeatMapChartBase({
+      data: emptyData,
+      domainValuesForColorScale: [],
+      rangeValuesForColorScale: [],
+    });
+    expect(instance).toBeDefined();
+    const result = instance._getStringFormattedDate('100000000');
```
Comment (AtishayMsft) [inline packages/react-charting/UnitTests/HeatMapChartUT.test.tsx:489]: This test case is interesting. How is this returning Jan/02

File: packages/react-charting/UnitTests/HeatMapChartUT.test.tsx [hunk 4]
```diff
@@ -0,0 +1,676 @@
...
+  test('Should return proper xAxis data points for numeric points', () => {
+    const instance = new HeatMapChartBase({
+      data: emptyData,
+      domainValuesForColorScale: [],
+      rangeValuesForColorScale: [],
+    });
+    expect(instance).toBeDefined();
+    instance._xAxisType = XAxisTypes.NumericAxis;
+    const result = instance._getXAxisDataPoints({ 100: '1', 30: '1' });
+    expect(result[0]).toEqual('30');
```
Comment (AtishayMsft) [inline packages/react-charting/UnitTests/HeatMapChartUT.test.tsx:538]: shoudln't result[0] equal 100 and result[1] equal 30

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

File: packages/react-charting/UnitTests/HeatMapChartUT.test.tsx (hunk 2 above)
Comment (AtishayMsft) [reply packages/react-charting/UnitTests/HeatMapChartUT.test.tsx:223]: Similarly all other unit tests involving date will break based on timezones of other machines where they are run.

Comment (v-baambati) [reply packages/react-charting/UnitTests/HeatMapChartUT.test.tsx:223]: Moved respective tests into skipped block.

File: packages/react-charting/UnitTests/HeatMapChartUT.test.tsx (hunk 4 above)
Comment (v-baambati) [reply packages/react-charting/UnitTests/HeatMapChartUT.test.tsx:538]: returning data in sorted format.

File: packages/react-charting/UnitTests/HeatMapChartUT.test.tsx (hunk 3 above)
Comment (v-baambati) [reply packages/react-charting/UnitTests/HeatMapChartUT.test.tsx:399]: While setting the time in milli seconds date is always considering January 1, 1970 GMT as current date.
    date.setTime(+point)

File: packages/react-charting/src/components/HeatMapChart/HeatMapChart.base.tsx [hunk 5]
```diff
@@ -356,28 +356,28 @@ export class HeatMapChartBase extends React.Component<IHeatMapChartProps, IHeatM
    * and un-highlight the rest of them
    * @param legendTitle
    */
-  private _onLegendHover = (legendTitle: string): void => {
+  private _onLegendHover(legendTitle: string): void {
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/HeatMapChart/HeatMapChart.base.tsx:359]: @srmukher Is this change still needed to run unit tests.

Comment (srmukher) [reply packages/react-charting/src/components/HeatMapChart/HeatMapChart.base.tsx:359]: Yes @AtishayMsft, this is required as "jest.spyOn(HeatMapChartBase.prototype as any, '_onLegendHover');" requires the function to be a classic non-arrow function

**Comment by AtishayMsft** [comment]
The change file is missing.

**Comment by v-baambati** [comment]
Hi @AtishayMsft added change file for this PR.

File: change/@fluentui-react-charting-56616556-b911-45f1-a97c-07bf990ebab8.json [hunk 6]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-charting-56616556-b911-45f1-a97c-07bf990ebab8.json:1]: ## [🕵 fluentuiv8 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=30212&commitId=e25c7d1b5a5fba8b226dbb30d3465e27963d735a&env=prod&clientType=FLUENTUI)


  <details><summary><b>react-charting-AreaChart</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|react-charting-AreaChart.Custom Accessibility.default.chromium.png|11| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

Comment (AtishayMsft) [reply change/@fluentui-react-charting-56616556-b911-45f1-a97c-07bf990ebab8.json:1]: This change is screenshot is unrelated to this PR.

**Review by AtishayMsft — APPROVED** [review]
